libs/bytes/bytes.go

```diff
@@ -69,3 +69,24 @@ func (bz HexBytes) Format(s fmt.State, verb rune) {
 		s.Write([]byte(fmt.Sprintf("%X", []byte(bz))))
 	}
 }
+
+func (bz HexBytes) MarshalJSON() ([]byte, error) {
```

Comment:
We should treat this as an app-hash breaking change due to potentially wide-ranging impact.